### PR TITLE
feat(draw): new features, improvements and bug fixes

### PR DESCRIPTION
**Pull request by LAMM26**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines:https://github.com/infra-geo-ouverte/igo2/blob/master/.github/CONTRIBUTING.md#git-commit-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
https://github.com/infra-geo-ouverte/igo2/issues/655
https://github.com/infra-geo-ouverte/igo2/issues/649
https://github.com/infra-geo-ouverte/igo2/issues/646

**What is the new behavior?**
Issue 655:
- L'attribut id correspond à l'étiquette lorsque présente, correspond à l'id (simplifié) sinon.

Issue 649:
- Les sélecteurs de couleur ont été déplacés vers la gauche.
- Un délai de 250 ms a été ajouté avant l'apparition de la boîte de dialogue permettant de saisir une étiquette.
- Un bouton "Aucune étiquette" a été ajouté à la boîte de dialogue de saisie d'étiquette.
- La couleur de contour par défaut a été modifiée à rouge afin de distinguer une entité sélectionnée plus facilement.
- Les contours des entités dessinées ont été amincies afin de rendre une entité sélectionnée plus visible.
- Il est maintenant possible d'exporter les entités de type Cercle en polygones dans l'outil d'exportation.
- Ajout de raccourcis clavier lors du dessin (Esc ou c pour annuler le dessin, Backspace ou u pour enlever le dernier vertex, Enter ou f pour terminer le dessin).
- Autre modifications de l'interface dans l'outil Dessin.

Issue 646:
- Une entité peut être modifiée en déplaçant des arêtes ou des vertex.
- L'étiquette d'une entité peut être modifiée en double cliquant sur une entité et en entrant une nouvelle étiquette.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**

**Other information**:

### CODE REVIEW

File: packages/geo/src/lib/draw/shared/draw-style.service.ts [hunk 1]
```diff
@@ -10,160 +11,176 @@ import { MapService } from '../../map/shared/map.service';
...
+  }
+
+  getIcon() {
+    return this.icon;
+  }
+
+  createDrawingLayerStyle(feature, resolution, labelsAreShown?: boolean, icon?: string): OlStyle.Style {
+    let style;
+    let labelsAreOffset: boolean = false;
+    const proj = this.mapService.getMap().ol.getView().getProjection().getCode();
```
Comment (pelord) [inline packages/geo/src/lib/draw/shared/draw-style.service.ts:76]: maybe refer to the view controller.
this.mapService.getMap().viewController.getOlProjection().getCode()

File: packages/geo/src/lib/draw/shared/draw-style.service.ts [hunk 2]
```diff
@@ -10,160 +11,176 @@ import { MapService } from '../../map/shared/map.service';
...
+
+  setFillColor(fillColor: OlColor) {
+    this.fillColor = fillColor;
+  }
+
+  getStrokeColor(): OlColor {
+    return this.strokeColor;
+  }
+
+  setStrokeColor(strokeColor: OlColor) {
```
Comment (pelord) [inline packages/geo/src/lib/draw/shared/draw-style.service.ts:37]: validate if all the following method are used...

File: packages/geo/src/lib/import-export/import-export/import-export.component.html [hunk 3]
```diff
@@ -141,6 +140,14 @@ <h4>{{'igo.geo.importExportForm.exportNoLayersExportable' | translate}}</h4>
     </mat-slide-toggle>
   </div>
 
+  <div class="export-options mat-typography" *ngIf="form.value.format !== 'URL' && circlesArePresent">
```
Comment (pelord) [inline packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: This control is not useful. It is better if you directly store a polygon,

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts [hunk 4]
```diff
@@ -550,6 +549,23 @@ export class ImportExportComponent implements OnDestroy, OnInit {
         geomTypes = [{ geometryType: '', features: olFeatures }];
       }
 
+      if (data.circlesToPolygons) {
+        geomTypes = [];
```
Comment (pelord) [inline packages/geo/src/lib/import-export/import-export/import-export.component.ts:553]: this action could be done a the draw-component level. Always store a polygon.

refer to ol/geometry/polygon/fromCircle

File: packages/geo/src/lib/draw/draw/draw-popup.component.html [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+<div mat-dialog-content>
+  <p class="mat-typography">{{'igo.geo.draw.dialogInstruction' | translate}}</p>
+  <mat-form-field class="example-full-width">
+    <input
+      #input
+      matInput
+      placeholder="{{'igo.geo.draw.dialogTitle' | translate}}">
+  </mat-form-field>
+</div>
+<div mat-dialog-actions>
```
Comment (pelord) [inline packages/geo/src/lib/draw/draw/draw-popup.component.html:10]: Maybe keep the cancel button to really cancel the drawing.

**Review by pelord — COMMENTED** [review]
General comment : 
1- Split your differents modification into commit. issueX = 1 or more commit.  Your PR is too broad and it is hard to review.
2- Ensure the content is mobile friendly (
image
)
3- This is not a final review. A new review will be needed aftherward.

**Comment by pelord** [comment]
@LAMM26 Good job!!

**Comment by pelord** [comment]
@LAMM26 One more thing, when modifying the circle, show the result of th drag..... not only a the end of the drag

File: packages/geo/src/lib/import-export/import-export/import-export.component.html (hunk 3 above)
Comment (pelord) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: I see... you store a circle and export a polygon.... to let the circle editable... ok...

Comment (LAMM26) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: Yes, it was discussed with the team and it was determined that it was for the best to do it that way (give the user the option to transform in a polygon or keep the point with radius). If the circle is transformed in a polygon when the drawing is done, the circle is then a x sided polygon and you can't modify it. It really depends on the need of the user(s).

File: packages/geo/src/lib/draw/shared/draw-style.service.ts (hunk 2 above)
Comment (LAMM26) [reply packages/geo/src/lib/draw/shared/draw-style.service.ts:37]: > 1- Split your differents modification into commit. issueX = 1 or more commit. Your PR is too broad and it is hard to review.
Gonna take that into account in the future!

File: packages/geo/src/lib/import-export/import-export/import-export.component.html (hunk 3 above)
Comment (pelord) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: This make no sense to let a point + a radius been exported!!

Comment (pelord) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: It's ok for me to store a circle.. but when exporting, do not let the user export a point+radius!

File: packages/geo/src/lib/draw/shared/draw-style.service.ts (hunk 1 above)
Comment (pelord) [reply packages/geo/src/lib/draw/shared/draw-style.service.ts:76]: or getMap().projection

Comment (pelord) [reply packages/geo/src/lib/draw/shared/draw-style.service.ts:76]: image

File: packages/geo/src/lib/draw/draw/draw-popup.component.html (hunk 5 above)
Comment (LAMM26) [reply packages/geo/src/lib/draw/draw/draw-popup.component.html:10]: image

There never was a Cancel button in the dialog box (see image above)? The user could click outside the box to close it (it acted like the user said 'no label') so I just added a dedicated button for clarity. The drawing is technically already 'finished' by that point (the drawend event fires and the dialog box appears). But something I forgot to mention is that I added some keyboard shortcuts to abort the current drawing (while drawing a feature) (Esc or 'c').

**Comment by LAMM26** [comment]
> @LAMM26 One more thing, when modifying the circle, show the result of th drag..... not only a the end of the drag

File: packages/geo/src/lib/import-export/import-export/import-export.component.html (hunk 3 above)
Comment (PhilippeLafreniere18) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: Can it be useful for certain exports format ? Maybe if someone wants a CSV file with the center coordinates and radius length ? If yes, i think the best would be to set default export to polygon, but allow the user to export a circle with radius if he really wants to

Comment (LAMM26) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: The coordinates of the center (lon, lat) of a circle and its radius are kept as attributes for every format. Furthermore, the coordinates for point-type features are also saved as attributes. That way, CSVs contain more info and can be more useful in GIS applications.

Comment (PhilippeLafreniere18) [reply packages/geo/src/lib/import-export/import-export/import-export.component.html:143]: Good ! In that case, i would say always export as polygon

**Review by pelord — APPROVED** [review]

**Pull request merged**